### PR TITLE
DCOS-43779: Change dist output

### PR DESCRIPTION
**Pull request by Poltergeist**
This changes the dist output to be like:

```
dist/
 |- index.html
  \- assets/
      |- index.js
      \- ...
```

to test run `npm run build` and check the output. To test it against a cluster run `node_modules/.bin/http-server -sp 4200 ./dist --proxy-secure=false -P <CLUSTER_URL>` and visit http://localhost:4200

### CODE REVIEW

**Comment by GeorgiSTodorov** [comment]
Did we reach a conclusion on whether we want to do this?

**Comment by nLight** [comment]
@GeorgiSTodorov good question. What would be your concern regarding this change?

**Comment by nLight** [comment]
@GeorgiSTodorov here's the reasoning: https://mesosphere.slack.com/archives/C03UXUVH7/p1539778932000100
TL;DR that would simplify and harden the Admin router setup when we start serving UI static from the update service

**Comment by GeorgiSTodorov** [comment]
Yeah, I read that, but I was wondering if we as a team agreed to do it, because that discussion seems pretty open.

**Comment by GeorgiSTodorov** [comment]
@Poltergeist is in vacation this week, should we look into this while he is gone or wait until he is back?

**Comment by nLight** [comment]
Yes, we should take this one over

**Comment by GeorgiSTodorov** [comment]
@Poltergeist is in vacation this week, do we want to finish this quickly or can we wait until he is back?

**Comment by nLight** [comment]
Yes, we should take this one over

**Comment by nLight** [comment]
Yes, we should take this one over

**Comment by GeorgiSTodorov** [comment]
@Poltergeist is in vacation this week, do we want to finish this quickly or can we wait until he is back?

**Comment by GeorgiSTodorov** [comment]
@Poltergeist is in vacation this week, do we want to finish this quickly or can we wait until he is back?

**Comment by GeorgiSTodorov** [comment]
@Poltergeist is in vacation this week, do we want to finish this quickly or can we wait until he is back?

**Comment by GeorgiSTodorov** [comment]
Sorry, github glitched.

File: webpack/webpack.config.js [hunk 1]
```diff
@@ -123,7 +124,7 @@ module.exports = {
           {
             loader: "file-loader",
             options: {
-              name: "./[hash]-[name].[ext]"
+              name: "./assets/[hash]-[name].[ext]"
```
Comment (juliangieseke) [inline webpack/webpack.config.js]: for some reason this doesnt catch `sprite.svg` file 🤔

Comment (nLight) [reply webpack/webpack.config.js]: yes, look at `svg-compiler-plugin.js` 😩 I managed to move it into `assets` by setting `compilation.assets["assets/sprite.svg"]` but then UI wouldn't show the icons.

Comment (juliangieseke) [reply webpack/webpack.config.js]: ah, thanks! we also have to change `./src/js/index.js@L39` to load the sprite from assets… 🤷‍♂️

Comment (nLight) [reply webpack/webpack.config.js]: It didn't resolve the issue for me

Comment (juliangieseke) [reply webpack/webpack.config.js]: hm, what exactly are you testing? 🤔

Comment (nLight) [reply webpack/webpack.config.js]: I do the build and then test against `http-server`. I check if all the assets are in `/dist/assets` and that the UI still has all the styles, images and icons.

Comment (juliangieseke) [reply webpack/webpack.config.js]: gotcha, it tries to load images from `http://127.0.0.1:4200/assets/assets/…` 😞

**Comment by nLight** [comment]
Pushed my `wip`

**Comment by GeorgiSTodorov** [comment]
System tests in `services/test-apps.js` are failing.

**Comment by juliangieseke** [comment]
will have another look now

**Comment by juliangieseke** [comment]
hm, no idea how to tell all the loaders to use different folders… with the current setup css files, which are in `assets` folder, reference their images also with `assets/…` which leads to a combindes path of `/assets/assets/…`… I tried different things but wasnt able to tell webpack (and all our quicky scripts, e.g. `./scripts/inline-bootstrap-css.js`) to correctly set path.

I'd recomment to skip this and instead use the correct config in AdminRouter…

**Comment by brandonc** [comment]
I took this approach: 

https://github.com/dcos/dcos-ui/pull/3405

**Comment by brandonc** [comment]
I think we should go with this approach https://github.com/dcos/dcos-ui/pull/3405

**Pull request closed without merging**